### PR TITLE
Add svgeez addon

### LINKED ISSUE #425: Add svgeez addon
## Why

As we use the https://github.com/jgarber623/svgeez for most of web project

## Who Benefits?

Developers

```[tasklist]
### Tasks
```

### PR DESCRIPTION
**Pull request by mosharaf13**
close #425

## What happened 👀

- Add svgeez as installable add-on while this template is being installed

## Insight 📝

- Adds `svgeez` as a gem in Gemfile

## Proof Of Work 📹
Screen Shot 2023-07-13 at 5 31 47 PM
Screen Shot 2023-07-13 at 5 33 04 PM

### CODE REVIEW

**Comment by andyduong1920** [comment]
Sorry @mosharaf13 I updated the original issue, which we are using https://github.com/jgarber623/svgeez gem, you can have a look for that on

- https://github.com/nimblehq/cx-web/blob/release/poc/Gemfile#L64
- https://github.com/nimblehq/cx-web/blob/release/poc/bin/svg-sprite
- https://github.com/nimblehq/cx-web/blob/release/poc/.github/wiki/Add-new-svg-icon.md

**Comment by mosharaf13** [comment]
> Sorry @mosharaf13 I updated the original issue, which we are using https://github.com/jgarber623/svgeez gem, you can have a look for that on
> 
> * https://github.com/nimblehq/cx-web/blob/release/poc/Gemfile#L64
> * https://github.com/nimblehq/cx-web/blob/release/poc/bin/svg-sprite
> * https://github.com/nimblehq/cx-web/blob/release/poc/.github/wiki/Add-new-svg-icon.md

Hi @andyduong1920,

I have replaced svg-sprite with svgeez and updated the PR

**Review by malparty — APPROVED** [review]
Tests are failing, could you have a look please? 🙏

**Comment by andyduong1920** [comment]
Also to finalize the feature, we will need the `inline_svg` gem as well could you help to add that Gem too 🙏 https://github.com/nimblehq/cx-web/blob/1d9e3290b886d75215786189b38126e74a523ae1/Gemfile#L26

File: .template/addons/svgeez/template.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# frozen_string_literal: true
+
+apply '.template/addons/svgeez/Gemfile.rb'
```
Comment (andyduong1920) [inline .template/addons/svgeez/template.rb:3]: is this in the web project only @mosharaf13, for API variant, I think we don't need this gem

File: .template/addons/svgeez/.github/wiki/Add-new-svg-icon.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  ```
+
+## Use the new icon
+- Add the `icon-sprite.svg` file to the layout of the page
+  ```erb
+  <body>
+    <%= inline_svg_tag 'icon-sprite.svg' %>
+  </body>
+  ```
+- Use the `svg_tag` provided by the `SvgHelper` (app/helpers/svg_helper.rb) and provided `icon_id` matched with icon file name with prefix `icon-`:
```
Comment (andyduong1920) [inline .template/addons/svgeez/.github/wiki/Add-new-svg-icon.md:29]: @mosharaf13 let's double check that do we have the `app/helpers/svg_helper.rb` file in the existing template or not, if not, let's handle that file too 🙏

File: .template/addons/svgeez/.github/wiki/template.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+
+# frozen_string_literal: true
+
+use_source_path __dir__
+
+copy_file 'Add-new-svg-icon.md', '.github/wiki/Add-new-svg-icon.md'
```
Comment (andyduong1920) [inline .template/addons/svgeez/.github/wiki/template.rb:6]: 1 thing, the `Github addon is an optional addon` -> `install_addon_prompt('Github Action and Wiki')` so the `.github/wiki` might not exist if the user does not select the Github Addon, I think here we need to do something like check if the directory existing or not before copy, what do you think @mosharaf13

**Review by andyduong1920 — APPROVED** [review]

File: .template/addons/svgeez/.github/wiki/template.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
+
```
Comment (olivierobert) [inline .template/addons/svgeez/.github/wiki/template.rb:1]: This extra line break should be removed

**Review by olivierobert — APPROVED** [review]
Just one minor code style issue

File: .template/addons/svgeez/bin/svg-sprite [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+# Usage
+#  -s --source:	Path to the folder of source SVGs (defaults to ./_svgeez).
+#  -d --destination:	Path to the destination file or folder (defaults to ./svgeez.svg)
+#  --with-svgo:	Optimize SVG sprite file with SVGO
+
+# Generate the sprite file which includes the `icon-` prefix
+bin/bundle exec svgeez build --source app/assets/images/icons/ --destination app/assets/images/icon.svg --with-svgo
+
+# Rename the icon.svg to icon-sprite.svg
+mv app/assets/images/icon.svg app/assets/images/icon-sprite.svg
```
Comment (tyrro) [inline .template/addons/svgeez/bin/svg-sprite:15]: We can combine these two steps together now as `svgeez` now [supports prefix flag](https://github.com/jgarber623/svgeez#options-and-defaults)

```suggestion
# Generate the sprite file which includes the `icon-` prefix
bin/bundle exec svgeez build --prefix icon --source app/assets/images/icons/ --destination app/assets/images/icon-sprite.svg --with-svgo
```

File: .template/addons/svgeez/Gemfile.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+# SVG Sprite
+insert_into_file 'Gemfile', after: /gem 'bcrypt'.*\n/ do
+  <<~RUBY
+
+    # Svgeez
+    gem 'svgeez' # Gem for generating an SVG sprite from a folder of SVG icons.
```
Comment (tyrro) [inline .template/addons/svgeez/Gemfile.rb:8]: I think we only need the `svgeez` gem to generate the sprite file,  so we only need it during development  - so could we move it inside the `development`  group?

**Review by tyrro — CHANGES_REQUESTED** [review]

File: .template/addons/svgeez/.github/wiki/template.rb [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+if Dir.exist?('.github/wiki')
+  use_source_path __dir__
+
+  copy_file 'Add-new-svg-icon.md', '.github/wiki/Add-new-svg-icon.md'
+
+  # SVG Sprite
+  insert_into_file '.github/wiki/_Sidebar.md', after: /## Operations.*\n/ do
+    <<~RUBY
+
+      - [[Add new svg icon]]
```
Comment (tyrro) [inline .template/addons/svgeez/.github/wiki/template.rb:13]: What do you think of renaming it to `Managing SVG icons` as the wiki is about managing the icons?

File: .template/addons/svgeez/.github/wiki/template.rb (hunk 4 above)
Comment (mosharaf13) [reply .template/addons/svgeez/.github/wiki/template.rb]: Thanks. Fixed in 5685935937d6e52e31976dd93d9506de59105ca1

File: .template/addons/svgeez/.github/wiki/template.rb (hunk 7 above)
Comment (mosharaf13) [reply .template/addons/svgeez/.github/wiki/template.rb:12]: Yeah. This title is more relevant 👍🏼

File: .template/addons/svgeez/Gemfile.rb (hunk 6 above)
Comment (malparty) [reply .template/addons/svgeez/Gemfile.rb:8]: As I see it is now ` after: /gem 'danger'.*`, I assume this one is resolved :)

Comment (mosharaf13) [reply .template/addons/svgeez/Gemfile.rb:8]: Yeah. Fixed in 624b044431f780bd3bdc693bc378ae891eea1471 🙏🏼

File: .template/addons/svgeez/.github/wiki/template.rb [hunk 8]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+if Dir.exist?('.github/wiki')
+  use_source_path __dir__
+
+  copy_file 'Add-new-svg-icon.md', '.github/wiki/Add-new-svg-icon.md'
```
Comment (tyrro) [inline .template/addons/svgeez/.github/wiki/template.rb:6]: As we now have renamed the section,  let's  rename the file as well 🙌🏿 

```suggestion
  copy_file 'Managing-SVG-Icons.md', '.github/wiki/Managing-SVG-Icons.md'
```

File: .template/addons/svgeez/.github/wiki/template.rb [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+
+  # SVG Sprite
+  insert_into_file '.github/wiki/_Sidebar.md', after: /## Operations.*\n/ do
+    <<~RUBY
+
```
Comment (tyrro) [inline .template/addons/svgeez/.github/wiki/template.rb:11]: we can remove this empty line,  no?

File: .template/addons/svgeez/Gemfile.rb [hunk 10]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+# SVG Sprite
+insert_into_file 'Gemfile', after: /gem 'danger'.*\n/ do
+  <<~RUBY
+
+    # Svgeez
+    gem 'svgeez' # Gem for generating an SVG sprite from a folder of SVG icons.
+    gem 'inline_svg' # Use Inline SVG for styling SVG with CSS
```
Comment (tyrro) [inline .template/addons/svgeez/Gemfile.rb:9]: sorry if the [previous comment](https://github.com/nimblehq/rails-templates/pull/445/files#r1269099739) was a bit unclear,  `svgeez` is needed only during development ✅ but `inline_svg` requires to be in production as we'll be using `inline_svg_tag` inside the application.

**Review by tyrro — CHANGES_REQUESTED** [review]

File: .template/addons/svgeez/Gemfile.rb (hunk 10 above)
Comment (tyrro) [reply .template/addons/svgeez/Gemfile.rb:9]: In a word:
-  `gem 'svgeez'`  -  development group
-   `gem 'inline_svg'` -  production group(the way you implemented in https://github.com/nimblehq/rails-templates/pull/445/commits/c0734c04a9e477ba39580d1811217dbbc4c432b3)

File: .template/addons/svgeez/.github/wiki/template.rb (hunk 8 above)
Comment (mosharaf13) [reply .template/addons/svgeez/.github/wiki/template.rb]: Thanks. Fixed in b21c85ca605d162ece23e0bbcc2a89f9d1eb8457

File: .template/addons/svgeez/.github/wiki/template.rb (hunk 9 above)
Comment (mosharaf13) [reply .template/addons/svgeez/.github/wiki/template.rb]: Thanks. Fixed in 1f5197abc6211ba7ac258c8fd69e339f7943486e

Comment (mosharaf13) [reply .template/addons/svgeez/.github/wiki/template.rb]: Thanks. Removed in 1f5197abc6211ba7ac258c8fd69e339f7943486e

File: .template/addons/svgeez/Gemfile.rb (hunk 10 above)
Comment (mosharaf13) [reply .template/addons/svgeez/Gemfile.rb:13]: Thanks. Fixed in 3b2dd797edfa8d0efad4e54a6c458f178e1c859a

**Review by tyrro — APPROVED** [review]

**Pull request merged**